lib/Onyx.js

```diff
@@ -1189,7 +1226,7 @@ function merge(key, changes) {
                 const updatePromise = broadcastUpdate(key, modifiedData, hasChanged, 'merge');
 
                 // If the value has not changed, calling Storage.setItem() would be redundant and a waste of performance, so return early instead.
-                if (!hasChanged) {
+                if (!hasChanged || ongoingClear) {
```

Comment:
Looks like the second operand never works here, because of a condition above, where you already checked `ongoingClear` equals `true`.

Reply:
This one might, because the `ongoingClear` might be false when the merge was called, but true by the time the `get` finishes executing